### ISSUE TITLE
PHP notice wp_cache_flush_runtime was called incorrectly

### ISSUE THREAD

**Issue by NataliaDrause** [wp-media/wp-rocket#5589]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version - 3.12.2.1
- Used the search feature to ensure that the bug hasn’t been reported before - yes

**Describe the bug**

The following PHP notice is being logged every minute:

`PHP Notice: Function wp_cache_flush_runtime was called <strong>incorrectly</strong>. Your object cache implementation does not support flushing the in-memory runtime cache. Please see <a href="https://wordpress.org/support/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.1.0.) in /home/customer/www/besttechnologyinc.com/public_html/wp-includes/functions.php on line 5835
`

This is happening when we call the function here: https://github.com/wp-media/wp-rocket/blob/13d1000311183a247d6b14f75ac780e0d93f8bd6/inc/Dependencies/ActionScheduler/classes/ActionScheduler_QueueRunner.php#L187

As per @vmanthos investigation:

> This function was introduced in WordPress 6.0 and Action Scheduler started using it since 3.5.1.
> AS needs to check if it's possible to clear the runtime cache here:
> [https://github.com/wp-media/wp-rocket/blob/5ae181cb176c520f7fc9990cd8f5c9e9f6281d5[…]dencies/ActionScheduler/classes/ActionScheduler_QueueRunner.php](https://github.com/wp-media/wp-rocket/blob/5ae181cb176c520f7fc9990cd8f5c9e9f6281d5e/inc/Dependencies/ActionScheduler/classes/ActionScheduler_QueueRunner.php#L187)
> maybe by using:
> ```
> if( wp_cache_supports( 'flush_runtime' ) ){
>     wp_cache_flush_runtime();
> }
> ```
> This is happening with other plugins using AS: https://wordpress.org/support/topic/numerous-php-warnings/

**To Reproduce**
Reported on customer's site that is using Memcached. WordPress version 6.1.1

**Expected behavior**
No notices should appear in error logs.

**Additional context**
Slack thread: https://wp-media.slack.com/archives/C43T1AYMQ/p1668790407001309

Ticket: https://secure.helpscout.net/conversation/2068968933/381875?folderId=3864740
Possibly related: https://secure.helpscout.net/conversation/2071847296/382555?folderId=3864740

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by alfonso100** [comment]
related https://secure.helpscout.net/conversation/2082026048/385661?folderId=2683093

**Comment by DahmaniAdame** [comment]
The warning comes from the following change on WordPress introduced in October 2022: 

https://github.com/WordPress/WordPress/commit/f821fa413ece0949a2593455300e0d8f32198ff4

We need to check first if the current Object cache supports `flush_runtime` and bail if it doesn't, as running https://github.com/wp-media/wp-rocket/blob/13d1000311183a247d6b14f75ac780e0d93f8bd6/inc/Dependencies/ActionScheduler/classes/ActionScheduler_QueueRunner.php#L181-L206 will have no effect, even with the fallback. 

If this test passes, we can then check for `wp_cache_flush_runtime()` and go through with the logic.

**Comment by DahmaniAdame** [comment]
A potential enhancement on SiteGround's case when the SG Optimizer is used. 

They have a method to flush Memcached cache when needed and if their Object Cache implementation is used: 

```
if ( Options::is_enabled( 'siteground_optimizer_enable_memcached' ) ) {
    $siteground_optimizer_loader->supercacher->flush_memcache();
}
```

**Comment by DahmaniAdame** [comment]
Happening on WP Engine as well.

**Comment by DahmaniAdame** [comment]
FYI @piotrbak - This was fixed on ActionScheduler 3.6.1. 
https://github.com/woocommerce/action-scheduler/blob/3.6.1/classes/ActionScheduler_QueueRunner.php#L193-L194

**Comment by piotrbak** [comment]
It's fixed in the AS itself
